dGame/dBehaviors/Behavior.h

```diff
@@ -88,5 +88,5 @@ class Behavior
 	 */
 
 	explicit Behavior(uint32_t behaviorId);
-	virtual ~Behavior();
+	virtual ~Behavior() = default;
```

Comment:
If the destructor is being defaulted, it should just be removed in order to avoid the weird side effects of having an explicitly defined destructor

Reply:
Ah. Right. It's virtual. Hm... I need to check if a defaulted destructor needs to be declared virtual or not. I think you might actually be right to keep it in this case.

Reply:
It might be worthwhile to explicitly define the move constructors as defaulted so they aren't implicitly deleted, then. I don't know if we actually make use of any move optimizations here, but some classes will, like vector, and it's good to future-proof even if we don't.

Reply:
Behaviors themselves are never moved or copied, they are always a pointer, a copy or move could be defined but they will never be used since the ptr will be copied, not the data.  Just going foe the low hanging fruits here since they didn't require much modification (it's very clear to me this was c# code originally ported to c++)

Reply:
I went ahead and did it myself then. Feel free to revert if you think it shouldn't be there.

Reply:
im not going to use time reverting them, but those will never be used because Behaviors are never used in that manner as I stated above.  whether those are defined or not will make zero difference since they wont be used